### PR TITLE
Get content and Implement search functionality in new ScholarX Archive page

### LINKED ISSUE #1413: Get content and Implement functionalities in new ScholarX Archive page
**Is your feature request related to a problem? Please describe.**
We implemented the initial UI for the ScholarX Archive page in #1388. Now all the content (data) and the functionalities (filtration) should be completed.

**Describe the solution you'd like**
Get the content via API and display it in the archive section. Implement filter funcitons.

**Describe alternatives you've considered**
None

**Additional context**
Screenshot 2022-11-23 at 01 53 31

**Are you working on it?**
Yes

### PR DESCRIPTION
**Pull request by Madhawa97**
## Purpose

The purpose of this PR is to fix #1413 

## Goals

This will add functionality to get the content via API and display it in the archive section. Filter by search feature is also added.

## Approach

- Got the content from a new API. The API includes all mentee and mentor data in 2019, 2020, and 2021 programs.
- Find and assign the mentees of each mentor using an algorithm. 
- Display mentors using mustache template.
- Add the filter by search functionality

### Screenshots
Screenshot 2022-11-23 at 02 03 59

  
### Preview Link
https://pr-1414-sef-site.surge.sh/scholarx/archive/

##  Checklist
- [x] I have read and understood the [development best practices](https://handbook.sefglobal.org/organisation/engineering-team#development-best-practices) guidelines
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas

## Related PRs
 
#1388 

## Learning

https://stackoverflow.com/questions/9707397/making-a-div-vertically-scrollable-using-css

### CODE REVIEW

**Comment by Madhawa97** [comment]
No previews with Surge for Draft PRs?

**Comment by YoshithaRathnayake** [comment]
> No previews with Surge for Draft PRs?

I think no

File: scholarx/archive/styles.css [hunk 1]
```diff
@@ -27,3 +28,26 @@ hr {
...
+#profile-container table th {
+    text-align: center;
+}
+
+#profile-container td,
+#profile-container th {
+    border-top: 1px solid #eeeeee;
+    text-align: left;
+    padding: 8px;
+}
```
Comment (YoshithaRathnayake) [inline scholarx/archive/styles.css]: Add end of the line @Madhawa97

**Review by YoshithaRathnayake — DISMISSED** [review]

File: scholarx/archive/functions.js [hunk 2]
```diff
@@ -1,3 +1,60 @@
...
+
+    mentors2019 = data.mentor2019;
+    mentees2019 = data.mentee2019;
+    mentors2019.forEach(mentor => {
+        menteesOfMentor = []
+        mentees2019.forEach(mentee => {
+            if (mentee.mentor == mentor.name) {
+                menteesOfMentor.push(mentee.name);
+            }
+        });
```
Comment (anjula-sack) [inline scholarx/archive/functions.js]: I think it's better to move this to google script. The API should return the data the way we want so we don't have to manipulate it here

**Review by anjula-sack — CHANGES_REQUESTED** [review]

File: scholarx/archive/functions.js (hunk 2 above)
Comment (Madhawa97) [reply scholarx/archive/functions.js]: Sure Anjula. I will do it

File: scholarx/archive/styles.css (hunk 1 above)
Comment (Madhawa97) [reply scholarx/archive/styles.css]: Cool

File: scholarx/archive/functions.js (hunk 2 above)
Comment (Madhawa97) [reply scholarx/archive/functions.js]: Its done 🎉

File: scholarx/archive/functions.js [hunk 3]
```diff
@@ -1,3 +1,21 @@
...
+async function getData() {
+    const response = await fetch(data_url);
+    const data = await response.json();
+    return data;
+}
+
+async function loadData() {
+    
+    const payload = await getData();
+
```
Comment (anjula-sack) [inline scholarx/archive/functions.js]:
```suggestion
```

File: scholarx/archive/functions.js [hunk 4]
```diff
@@ -1,3 +1,21 @@
...
+const data_url = "https://script.google.com/macros/s/AKfycbw0TVyldiK5ijUxjLJkhrxHpxZpIjoeLytZqrvOlftYeHywn1hBEnL0aHGIS8hOFxPp/exec";
+
+async function getData() {
+    const response = await fetch(data_url);
+    const data = await response.json();
+    return data;
+}
+
+async function loadData() {
+
```
Comment (anjula-sack) [inline scholarx/archive/functions.js]:
```suggestion
```

**Review by anjula-sack — CHANGES_REQUESTED** [review]

**Review by anjula-sack — APPROVED** [review]
Good job! @Madhawa97

**Comment by Madhawa97** [comment]
Thank you @anjula-sack

**Pull request merged**